libnavigation-base/src/test/java/com/mapbox/navigation/base/utils/route/NavigationRouteExTest.kt

```diff
@@ -0,0 +1,178 @@
+package com.mapbox.navigation.base.utils.route
+
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.testing.FileUtils
+import io.mockk.every
+import io.mockk.mockk
+import junit.framework.Assert
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+private typealias BooleansProvider = () -> List<Boolean?>
+
+class NavigationRouteExTest {
+
+    @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+    @RunWith(Parameterized::class)
+    class RouteHasUnexpectedClosures(
+        private val description: String,
+        private val routeRaw: String,
+        private val snappingIncludeClosures: BooleansProvider?,
+        private val snappingIncludeStaticClosures: BooleansProvider?,
+        private val expectedHasUnexpectedClosures: Boolean,
+    ) {
+
+        companion object {
+            @JvmStatic
+            @Parameterized.Parameters(name = "{0}")
+            fun data(): List<Array<Any?>> =
```

Comment:
Do you test anywhere null booleans in closures list? I see only true/false.

Reply:
yeah, that make sense. Cut the issue https://github.com/mapbox/mapbox-navigation-android/issues/6312